pyresample/boundary/boundary_sides.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2023 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Define the BoundarySides class."""
+
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class BoundarySides:
+    """A class to represent the sides of an area boundary.
+
+    The sides are stored as a tuple of 4 numpy arrays, each representing the
+    coordinate (geographic or projected) of the vertices of the boundary side.
+    The sides must be stored in the order (top, right, left, bottom),
+    which refers to the side position with respect to the coordinate array.
+    The first row of the coordinate array correspond to the top side, the last row to the bottom side,
+    the first column to the left side and the last column to the right side.
+    Please note that the last vertex of each side must be equal to the first vertex of the next side.
+    """
+    __slots__ = ['_sides']
```

Comment:
Why is `__slots__` needed here?

Reply:
I remember @djhoese suggesting me to use `__slots__` whatever possible to speed up stuffs ... 
Do we want to use a `named_tuple` instead @mraspaud? Or just a class without `__slots__`?

Reply:
My thought on `__slots__` is that low-level immutable classes should use it to reduce memory footprint and speedup execution. My opinion is that it should be used anywhere there isn't a strong reason not to (subclassing gets harder, etc). I still prefer a namedtuple here.